### DISCUSSION TITLE
🚀 Feature request: Add WYGWAM-style options such as "restrict allowed HTML" toggle to RTE field

### DISCUSSION THREAD

**Discussion by aasun** [ExpressionEngine/ExpressionEngine#3557] [Feature Requests]
### Suggest an idea for ExpressionEngine

**Is your feature request related to a problem? Please describe.**
Looking to be able to have RTE fields get some enhanced (easy to enable) features similar to WYGWAM to be able to start to move to more "first-party" installs.  So, nothing is currently "broken", but adding these customizations via the JSON configurations is complex and still not able to be 100% WYGWAM replacement.  Some other "common" configurations would be great to be implemented into a WYGWAM-style interface for the RTE field to not have to relay on the JSON configuration for EVERYTHING.  Or, better yet, just present all the Redactor or CK configuration options in a dropdown to add, similar to how WYGWAM configuration interface works.

**Describe the solution you'd like**
Add a "Restrict Allowed HTML" toggle. When "on" automatically prevent html that is not an allowed format or style to be pasted in to the field. Also add the configurable options to a "selectable" dropdown to the RTE field configuration.  This will not require that a user knows or writes the correct form of JSON structures (which is fiddly) to enable/disable features.  The JSON configuration can still be used for extra custom needs.

If all the options for each editor are available as drop-downs, then the value for the setting is a structured field (just like WYGWAM), then the RTE field immediately becomes a direct replacement for WYGWAM and becomes s no-brainer.

**Comment by gamebreakers808** [comment]
For the RTE CK editor version, we really need the Source button. We also need the ability to set height and width for images and add a class. We have users heavily use the paste from Word as well.
